collector/library/pod.js

```diff
@@ -208,52 +232,30 @@ class Pod {
      * in sending it to Lodash at the first attempt.
      * 
      */
-    async submitDataToLogstash({ data, batch, packetId = '' }) {
-        const body = JSON.stringify({ message: data.toString() });
+    async submitDataToLogstash({ data, batch, }) {
+        const dataAsString = data.toString();
+        let lines;
+
+        lines = (this.remainder + dataAsString).split('\n');
+        this.remainder = '';
+
+        // Run on all lines but the last one!
+        for (let i = 0; i < lines.length - 1; i++) {
```

Comment:
since you have no good way of distinguishing between a connection that was closed because of a server error, gateway / proxy issue or otherwise, i now think it's a real bug to move on to the next file after the connection is closed.
the correct behavior after a connection is closed is to start again from the top of the method - we need to get the manifest, compare the number of bytes handled for any batch, find the current batch to download, and download only it. then next iteration we must download the manifest again,
in short, this loop here is a bug. you should only look for the first file that was never entirely retrieved and download only it.